Separate entity entries for CoRIM and CoMID

yogeshbdeshpande opened the pull request.

thomas-fossati commented on concise-mid-tag.cddl:
```diff
@@ -35,6 +34,34 @@ module-metadata-map = {
+$module-role /= module-licensor
+$module-role /= module-maintainer
+
+; module-tag-creator, person or org that created the containing CoMID tag
```
```suggestion
; person or org that created the CoMID tag
```

thomas-fossati commented on concise-mid-tag.cddl:
```diff
@@ -35,6 +34,34 @@ module-metadata-map = {
+; module-tag-creator, person or org that created the containing CoMID tag
+module-tag-creator = 1
+
+; module creator, person or org that created the module
```
```suggestion
; person or org that created the module
```

thomas-fossati commented on concise-mid-tag.cddl:
```diff
@@ -35,6 +34,34 @@ module-metadata-map = {
+; module creator, person or org that created the module
+module-creator = 2
+
+; module distributor, entity that furthers marketing, selling or distribution of module 
```
```suggestion
; entity that furthers marketing, selling or distribution of the module 
```

thomas-fossati commented on concise-mid-tag.cddl:
```diff
@@ -35,6 +34,34 @@ module-metadata-map = {
+; module distributor, entity that furthers marketing, selling or distribution of module 
+module-distributor = 3
+
+; module licensor, person or org that holds the right to issue a module license
```
```suggestion
; person or org that holds the right to issue the module license
```

thomas-fossati commented on concise-mid-tag.cddl:
```diff
@@ -35,6 +34,34 @@ module-metadata-map = {
+; module licensor, person or org that holds the right to issue a module license
+module-licensor = 4
+
+; module maintainer, person or org making updates to a module. To be used when entity is different than module creator
```
```suggestion
; person or org making updates to a module.
; To be used when different from the module creator
```

thomas-fossati commented on concise-mid-tag.cddl:
```diff
@@ -35,6 +34,34 @@ module-metadata-map = {
+  * $$module-entity-map-extension
+}
+
+$module-role /= module-tag-creator
+$module-role /= module-creator
+$module-role /= module-distributor
+$module-role /= module-licensor
+$module-role /= module-maintainer
```
```suggestion
$module-role /= comid.tag-creator
$module-role /= comid.creator
$module-role /= comid.distributor
$module-role /= comid.licensor
$module-role /= comid.maintainer
```

yogeshbdeshpande replied:
This expands nicely considering the base name of module-role 
so each module-role could be either module-tag-creator
module-creator etc. I felt it a bit more clearer! But happy to consider your suggestion if others in the review agree!

thomas-fossati replied:
I think we decided to always use the `comid.` prefix in cases like the above.  Instead of the `module-` infix, I'd rather use `role`, so for example: `comid.role-distributor`

yogeshbdeshpande replied:
ok, then better to stick with your original comment!

thomas-fossati commented on concise-mid-tag.cddl:
```diff
@@ -35,6 +34,34 @@ module-metadata-map = {
+$module-role /= module-distributor
+$module-role /= module-licensor
+$module-role /= module-maintainer
+
```
I think these should go to the comid-code-points.cddl?

yogeshbdeshpande replied:
Agree will change this!

thomas-fossati commented on corim-code-points.cddl:
```diff
@@ -0,0 +1,17 @@
+; unsigned-corim
+corim-id = 123 ; TBD
+tags = 97
+dependent-rims = 122; TBD
+
+; corim-meta
+signer = 98
+deny-id = 99
```
I think we should have a `corim.` prefix for these?

yogeshbdeshpande replied:
These are existing code points moved here, so this can be done as part of a file wide clean up?

thomas-fossati replied:
sure, I was suggesting we could do it now since we are moving them around alread

yogeshbdeshpande replied:
Raised Issue #22 to address this comment!

thomas-fossati commented on signed-corim.cddl:
```diff
@@ -14,13 +14,29 @@ protected-signed-corim-header = {
+  * $$module-entity-entry-extension
+}
+
+$rim-role /= corim-creator
+$rim-role /= corim-signer
```
```suggestion
$rim-role /= corim.creator
$rim-role /= corim.signer
```

thomas-fossati commented on signed-corim.cddl:
```diff
@@ -14,13 +14,29 @@ protected-signed-corim-header = {
+$rim-role /= corim-creator
+$rim-role /= corim-signer
+
+; corim creator
+corim-creator = 1
+
+; corim signer, person or org responsible for signing the CoRIM
+corim-signer = 2
```
these should go to the CoRIM codepoints file?

yogeshbdeshpande replied:
Yes, will do it !

nedmsmith reviewed (approved):
Looks good except for comment about xCoRIM. xCoRIM should have its own entity definition since the roles are not corim creator/signer but xcorim creator/signer.

henkbirkholz commented:
If I signed a CoRIM, I'd be rather surprised if there are other entities taking on the role of revoking my CoRIM next to me? What am I missing?

yogeshbdeshpande commented:
I agree with Henk, there should not be a separate person role for xCoRIM. Only CoRIM role should be authorised to execute xCoRIM.

Any other thoughts would love to hear that!

From: Henk Birkholz ***@***.***>
Sent: 13 April 2021 17:37
To: ietf-rats/ietf-corim-cddl ***@***.***>
Cc: Yogesh Deshpande ***@***.***>; Author ***@***.***>
Subject: Re: [ietf-rats/ietf-corim-cddl] Separate entity entries for CoRIM and CoMID (#21)


If I signed a CoRIM, I'd be rather surprised if there are other entities taking on the role of revoking my CoRIM next to me? What am I missing?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/ietf-rats/ietf-corim-cddl/pull/21#issuecomment-818878550>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AGKXIV4EB7RII4RLMYZIW33TIRXL3ANCNFSM423NZ23Q>.

IMPORTANT NOTICE: The contents of this email and any attachments are confidential and may also be privileged. If you are not the intended recipient, please notify the sender immediately and do not disclose the contents to any other person, use it for any purpose, or store or copy the information in any medium. Thank you.

nedmsmith commented:
The signer is employing the ‘corim-signer’ role to sign the manifest and employing the ‘xcorim-signer’ to revoke the manifest. The entity and roles are not conflated. But yes, it is weird if the entity that issued the manifest was different from the entity that revoked it.

It is possible with cert chains that the entity name is a root CA while the entities that possess signing and revoking roles are different branches of the key hierarchy. So, it could be different keys that sign the manifest vs. that sign the xcorim.

Keeping them as independent entity maps doesn’t artificially limit the flexibility while still allows for the common case. We are only saving paper by conflating them.

-Ned

From: Henk Birkholz ***@***.***>
Reply-To: ietf-rats/ietf-corim-cddl ***@***.***>
Date: Tuesday, April 13, 2021 at 9:42 AM
To: ietf-rats/ietf-corim-cddl ***@***.***>
Cc: "Smith, Ned" ***@***.***>, Review requested ***@***.***>
Subject: Re: [ietf-rats/ietf-corim-cddl] Separate entity entries for CoRIM and CoMID (#21)


If I signed a CoRIM, I'd be rather surprised if there are other entities taking on the role of revoking my CoRIM next to me? What am I missing?

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub<https://github.com/ietf-rats/ietf-corim-cddl/pull/21#issuecomment-818878550>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABPMCSGU4WW3CFI23FDH6GTTIRXL3ANCNFSM423NZ23Q>.

nedmsmith commented:
See reply in other thread.

From: Yogesh Deshpande ***@***.***>
Reply-To: ietf-rats/ietf-corim-cddl ***@***.***>
Date: Tuesday, April 13, 2021 at 9:44 AM
To: ietf-rats/ietf-corim-cddl ***@***.***>
Cc: "Smith, Ned" ***@***.***>, Review requested ***@***.***>
Subject: Re: [ietf-rats/ietf-corim-cddl] Separate entity entries for CoRIM and CoMID (#21)

I agree with Henk, there should not be a separate person role for xCoRIM. Only CoRIM role should be authorised to execute xCoRIM.

Any other thoughts would love to hear that!

From: Henk Birkholz ***@***.***>
Sent: 13 April 2021 17:37
To: ietf-rats/ietf-corim-cddl ***@***.***>
Cc: Yogesh Deshpande ***@***.***>; Author ***@***.***>
Subject: Re: [ietf-rats/ietf-corim-cddl] Separate entity entries for CoRIM and CoMID (#21)


If I signed a CoRIM, I'd be rather surprised if there are other entities taking on the role of revoking my CoRIM next to me? What am I missing?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/ietf-rats/ietf-corim-cddl/pull/21#issuecomment-818878550>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AGKXIV4EB7RII4RLMYZIW33TIRXL3ANCNFSM423NZ23Q>.

IMPORTANT NOTICE: The contents of this email and any attachments are confidential and may also be privileged. If you are not the intended recipient, please notify the sender immediately and do not disclose the contents to any other person, use it for any purpose, or store or copy the information in any medium. Thank you.


—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub<https://github.com/ietf-rats/ietf-corim-cddl/pull/21#issuecomment-818883164>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABPMCSAJPUUQRIR2PSAMHTTTIRYGRANCNFSM423NZ23Q>.

nedmsmith commented:
Yes. You should be familiar with the TCG spec on xcorim as the CDDL for xcorim is patterned after the revised corim CDDL. I think the xcorim CDDL is quite stale.

From: Yogesh Deshpande ***@***.***>
Reply-To: ietf-rats/ietf-corim-cddl ***@***.***>
Date: Tuesday, April 13, 2021 at 10:38 AM
To: ietf-rats/ietf-corim-cddl ***@***.***>
Cc: "Smith, Ned" ***@***.***>, Review requested ***@***.***>
Subject: Re: [ietf-rats/ietf-corim-cddl] Separate entity entries for CoRIM and CoMID (#21)


@yogeshbdeshpande commented on this pull request.

________________________________

In corim-code-points.cddl<https://github.com/ietf-rats/ietf-corim-cddl/pull/21#discussion_r612650677>:

> +corim-id = 123 ; TBD

+tags = 97

+dependent-rims = 122; TBD

+

+; corim-meta

+signer = 98

+deny-id = 99

These are existing code points moved here, so this can be done as part of a file wide clean up?

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub<https://github.com/ietf-rats/ietf-corim-cddl/pull/21#discussion_r612650677>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABPMCSGAEMHJN6MRK5NT4P3TIR6Q7ANCNFSM423NZ23Q>.

thomas-fossati commented:
what's going on with the latest comments?  there is an awful lot of email noise in them, how did that happen?

yogeshbdeshpande commented:
> Looks good except for comment about xCoRIM. xCoRIM should have its own entity definition since the roles are not corim creator/signer but xcorim creator/signer.

Yes, added xCoRIM CDDL Code Point file and added new entity-entry for xCoRIM to keep it separate from CoRIM

thomas-fossati commented on corim.cddl:
```diff
@@ -0,0 +1,657 @@
+concise-reference-integrity-manifest =
```
this is autogenerated, it must not be checked in
